src/pyspiffe/svid/jwt_svid.py

```diff
@@ -1,19 +1,156 @@
+                                provided as parameter
+        ValueError:             when the token is blank or cannot be parsed
+        BundleNotFoundError:    if the bundle for the trust domain of the spiffe id from the 'sub'
+                                cannot be found the jwt_bundle_source
+        AuthorityNotFoundError: if the authority cannot be found in the bundle using the value from
+                                the 'kid' header
+    """
+
+    @classmethod
+    def parse_and_validate(
+        cls, token: str, jwt_bundle: JwtBundle, audience: []
+    ) -> 'JwtSvid':
+
+        token_header = jwt.get_unverified_header(token)
+        signing_key = jwt_bundle.findJwtAuthority(token_header['kid'])
+
+        claims = jwt.decode(
+            token,
+            algorithms=token_header['alg'],
+            key=signing_key,
+            audience=audience,
+            options={
+                'verify_signature': True,
+                'verify_aud': True,
+                'verify_exp': True,
+                'require': cls._required_claims,
+            },
+        )
+
+        spiffe_ID = SpiffeId.parse(claims['sub'])
+        result = JwtSvid(spiffe_ID, claims['aud'], claims['exp'], claims, token)
+
+        return result
+
+    # If and when https://github.com/jpadilla/pyjwt/issues/599 is fixed, this can be simplified/removed.
+    @classmethod
+    def _validate_exp(cls, expiration_date: str) -> None:
+        expiration_date = int(expiration_date)
+        utctime = timegm(datetime.datetime.utcnow().utctimetuple())
+        if expiration_date < utctime:
+            raise JwtSvidError(SIGNATURE_EXPIRED_ERROR)
+
+    ###
+    ### Verifies if any item specified by audience is present in audience_clains.
+    ### Raises:
+    ###     JwtSvidError: in case none of the items specified by audience is present in audience_claims.
+    ###
+    @classmethod
+    def _validate_aud(cls, audience_claims: [], audience: []) -> None:
+        if not audience_claims and not audience:
+            raise ValueError(
+                INVALID_INPUT_ERROR.format(
+                    'audience_claims and audience cannot be empty'
+                )
+            )
+        try:
+            if not any(aud in audience_claims for aud in audience):
+                raise JwtSvidError(AUDIENCE_NOT_MATCH_ERROR)
+        except Exception:
+            raise JwtSvidError(AUDIENCE_NOT_MATCH_ERROR)
 
-    def __init__(self, spiffe_id: SpiffeId, private_key: str, chain: List[str]):
-        self.spiffe_id = spiffe_id
-        self.privateKey = private_key
-        self.chain = chain
+    ###
+    ### Validates payload for required claims (aud, exp, sub), signature expiration and audience.
+    ###
+    ### Raises:
+    ###     JwtSvidError: in case a required claim is not present in payload, token is expired or aud claimm does not match audience parameter.
+    ###
+    @classmethod
+    def _validate_claims(cls, payload: {}, audience: []) -> None:
```

Comment:
what is the difference between _validate_claims and _validate_aud ?

Reply:
_validate_claims validates all claims (aud, exp, sub), _validate_aud performs the validations for aud claim only.